### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by amartya-rs**
In this PR:
- added login and logout page
- user authentication

preview: https://deploy-preview-3--hh-tube.netlify.app/

### CODE REVIEW

File: src/components/top-navigation/TopNav.jsx [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+      <nav className="main-nav">
+         <Link to="/">
+            <span className="h5 brand">hhTube</span>
+         </Link>
+         {!authState.isLoggedIn ? (
             <BiUserCircle
                color="white"
                size="2.15rem"
+               title="login"
+               style={{ cursor: "pointer" }}
```
Comment (dev-prabhat) [inline src/components/top-navigation/TopNav.jsx:21]: you can use class to give style to react icons also

**Review by dev-prabhat — APPROVED** [review]

**Review by dev-prabhat — APPROVED** [review]
Code is clean and readable
UI is also looking good
Just given a small suggestion

File: src/components/top-navigation/TopNav.jsx (hunk 1 above)
Comment (amartya-rs) [reply src/components/top-navigation/TopNav.jsx:21]: Thanks for the suggestion, will work on it.

**Review by sanu29 — APPROVED** [review]
Good to merge.

**Pull request merged**